src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -154,18 +101,22 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
     switch (response.status)
     {
     case EMBER_ZCL_OTA_QUERY_STATUS_UPDATE_AVAILABLE: {
+        // TODO: Add a method to OTARequestorDriver used to report error condictions
         VerifyOrReturn(response.imageURI.HasValue(), ChipLogError(SoftwareUpdate, "Update is available but no image URI present"));
 
         // Parse out the provider node ID and file designator from the image URI
-        NodeId nodeId = chip::kUndefinedNodeId;
-        CharSpan fileDesignator;
-        CHIP_ERROR err = requestorCore->ParseBdxUri(response.imageURI.Value(), nodeId, fileDesignator);
+        NodeId nodeId                         = kUndefinedNodeId;
+        char fileDesignatorBuffer[kUriMaxLen] = { 0 };
+        MutableCharSpan fileDesignator(fileDesignatorBuffer, kUriMaxLen);
+        CHIP_ERROR err = requestorCore->mBdxDownloader->ParseBdxUri(response.imageURI.Value(), nodeId, fileDesignator);
```

Comment:
It would have been good for ParseBdxUri, which is logic-only and stateless against a BDX Downloader, to just be a separate API function in the name space. It reduces dependency chain and makes it more easy to use.